HPCC-12883 Remove socket fd max from select and epoll

mckellyln opened the pull request:
This PR uses poll() in place of select() (except on Windows where there is a known won't fix issue with WSAPoll()) AND this PR also uses an array of pointers into heap storage for the SelectItem list of descriptors - so there is no longer indexing by fd into an array (epfdtbl) and thus no longer any limit on the number or value of socket descriptors.  The change for using an array of pointers is referenced in JIRA issue HPCC-10235.  Both issues are fixed here because both relate to a max fd value.

@ghalliday and/or @jakesmith please review.

Signed-off-by: Mark Kelly mark.kelly@lexisnexis.com

jamienoss committed:
HPCC-14405 Redis Plugin - use cached connection for subscriptions

Signed-off-by: James Noss <james.noss@lexisnexis.com>

wangkx committed:
HPCC-15341 add abortby and abortTime for workunit

When a workunit is aborted, store abortby and abortTime.
Also return the abortby and abortTime in WsWorkunits.WUInfo.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

wangkx committed:
HPCC-14703 Add Cassandra based logging agent

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

jamienoss committed:
HPCC-15292 Redis Plugin - make connection caching optional

Signed-off-by: James Noss <james.noss@lexisnexis.com>

wangkx committed:
HPCC-14879 Support gzip option in soapcall

This fix enables gzip compress in soapcall request
and response. Also allows gzip in httpcall response.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

wangkx committed:
HPCC-15216 Add WsWorkunits.WUUpdateQueryEntry

For now, this method can only be used to edit query comments. We
may add more update as needed.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

ghalliday committed:
HPCC-15361 Use std::atomic for spin locks

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Update Visualization submodule to match 6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-15392 Remove non-functional crcResources option in Roxie

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8476 from wangkx/h15216

HPCC-15216 Add WsWorkunits.WUUpdateQueryEntry

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8502 from wangkx/h14879a

HPCC-14879 Support gzip option in soapcall

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8472 from wangkx/h14168

HPCC-15341 add abortby and abortTime for workunit

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8515 from richardkchapman/crcResources

HPCC-15392 Remove non-functional crcResources option in Roxie

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-15445 Errors building master on OSX

Checking a reference against NULL generates a fatal warning on Clang.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8550 from richardkchapman/osxbuild

HPCC-15445 Errors building master on OSX

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-15445 Errors building master on OSX

Checking a reference against NULL generates a fatal warning on Clang.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8551 from richardkchapman/osxbuild2

HPCC-15445 Errors building master on OSX

richardkchapman committed:
HPCC-14459 Job queues should not ignore queued items from closed clients

Makes some sense for Thor, since if eclagent has closed the job will fail. But
not otherwise.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-15447 Move a few static variables out of functions

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

wangkx committed:
HPCC-15454 Filter out duplicated schemas for WU Results XML

Duplicated schemas cause WsECL QueryOutputView to show duplicated
results. In this fix, duplicated schemas are filtered out before
using them.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

Commit:
HPCC-15451 Add c++11 move semantics to string classes

Signed-off-by: Gleb Aronsky <gleb.aronsky@lexisnexis.com>

ghalliday committed:
Merge pull request #8552 from richardkchapman/validateitemsessions

HPCC-14459 Job queues should not ignore queued items from closed clients

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-15392 Remove non-functional crcResources option in Roxie

Previous commit gave build errors if unit testing enabled.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jamienoss committed:
HPCC-12635 Add example plugin

Signed-off-by: James Noss <james.noss@lexisnexis.com>

shamser committed:
HPCC-13826 Extension of index_field[1..n]=field to IN

Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

ghalliday committed:
Merge pull request #8562 from richardkchapman/roxie-build

HPCC-15392 Remove non-functional crcResources option in Roxie

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-15404 Optimize roxie standalone queries

Step 1 - moving ownership of the threadpools into the RoxieQueue objects. This
commit should not change any behaviour.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-15404 Optimize roxie standalone queries

Start worker threads on demand.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-15486 Remove unused jmalloc code

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-11367 review and clean up use of spinlocks

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-15362 Remove other uses of atomic_t from jlib

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-14879 Fix windows compile regression

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #8590 from ghalliday/issue15486

HPCC-15486 Remove unused jmalloc code

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-15496: Refactor DataBuffer code

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #8598 from ghalliday/issue14879

HPCC-14879 Fix windows compile regression

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jamienoss committed:
HPCC-15499 Redis Plugin: add redis command INCRBY

Signed-off-by: James Noss <james.noss@lexisnexis.com>

ghalliday committed:
HPCC-11367 Add spinlock back into constructor foolowing review

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-15496 Changes following code review

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
HPCC-15404 Optimize roxie standalone queries

Address race condition questions, and default to prestarting threads when
deployed via ConfigMgr. Standalone Roxie programs will NOT typically load
a RoxieTopology file and will default to not prestarting threads.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8611 from jamienoss/issue15292-redis-plugin-make-connection-caching-optional-atomic-crit

HPCC-15292 Redis Plugin - make connection caching optional

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #8606 from ghalliday/issue15496

HPCC-15496: Refactor DataBuffer code

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8559 from ghalliday/issue15447

HPCC-15447 Move a few static variables out of functions

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8582 from wangkx/h14703

HPCC-14703 Add Cassandra based logging agent

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-15512 Add a csv output option to daliadmin stats

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
Merge pull request #8583 from richardkchapman/roxie-lite

HPCC-15404 Optimize roxie standalone queries

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #8591 from ghalliday/issue11367

HPCC-11367 review and clean up use of spinlocks

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8615 from jamienoss/issue15499-redis-plugin-add-INCRBY-rebase

HPCC-15499 Redis Plugin: add redis command INCRBY

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

jakesmith committed:
HPCC-15525 Ensure index parts are not reloaded in child queries

This commit also tidies up some of the common handling of index
parts.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #8619 from ghalliday/issue15512

HPCC-15512 Add a csv output option to daliadmin stats

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

wangkx committed:
HPCC-15513 Fix missing Build Version in ZAP report

In the existing WsWorkunits.WUCreateZAPInfo, both ESP Build Version
and ESP Network Address are added to ZAP report based on the values
inside the WUCreateZAPInfo request (which are forwarded by Create
ZAP Report form). When an ECLWU ZAP report is not created using the
form, both ESP Build Version and ESP Network Address are empty in
the report. In this fix, the code is added to read those values
into the report.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

richardkchapman committed:
Merge pull request #8561 from ghalliday/issue15361

HPCC-15361 Use std::atomic for spin locks

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8638 from wangkx/h15513

HPCC-15513 Fix missing Build Version in ZAP report

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-15535 USE_CASSANDRA changes for cassandralogagent target

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

jamienoss committed:
HPCC-15542 Regression from HPCC-15499

Signed-off-by: James Noss <james.noss@lexisnexis.com>

richardkchapman committed:
Merge pull request #8646 from jamienoss/issue15542-regression-from-HPCC-15499

HPCC-15542 Regression from HPCC-15499

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8641 from Michael-Gardner/HPCC-15535

HPCC-15535 USE_CASSANDRA changes for cassandralogagent target

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8558 from garonsky/add_move_2

HPCC-15451 - Add c++11 move semantics

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-14142 Roxie could avoid exists() call in fileUpToDate

A slight reduction in load on dafilesrv...

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8652 from richardkchapman/roxie-exists

HPCC-14142 Roxie could avoid exists() call in fileUpToDate

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
HPCC-15551 Reduce false positive suggestions for adding independent

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
HPCC-8116 More compiler warning fixes

More components can now be built with -Wall set.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-15572 Allow the parser state table to dynamically expand

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

jakesmith committed:
HPCC-15525 Minor changes following review

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
Merge pull request #8637 from jakesmith/hpcc-15525

HPCC-15525 Ensure index parts are not reloaded in child queries

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #8666 from ghalliday/issue15572

HPCC-15572 Allow the parser state table to dynamically expand

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-15565 Fix problems with inconsistent expression folding

If something is folded when binding a parameter it should be folded
when a the ECL is parsed, otherwise it can lead to inconsistencies.

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
HPCC-8116 More compiler warning fixes

Address concerns spotted in review.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8665 from ghalliday/issue15551

HPCC-15551 Reduce false positive suggestions for adding independent

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8678 from ghalliday/issue15565

HPCC-15565 Fix problems with inconsistent expression folding

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8657 from richardkchapman/warn-kill

HPCC-8116 More compiler warning fixes

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #8644 from jamienoss/issue12635-plugin-embed-example

HPCC-12635 Add example plugin

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

rpastrana committed:
HPCC-15619 Enhance WsESDLConfig getESDLBinding method

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
Merge pull request #8560 from wangkx/h15314_master

HPCC-15454 Filter out duplicated schemas for QueryOutputView

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

gaojx committed:
HPCC-15657: ESPLogging Agent: Filters Type should be enumeration, instead of free text

richardkchapman committed:
HPCC-15494 Workunit results aren't cleared when WU is cloned

If EXTEND specified, this can be an issue (if not they typically end up
overwritten when WU is rerun).

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

wangkx committed:
HPCC-15664 Add gzip header to http/soapcall test result key files

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

jakesmith committed:
HPCC-15666 Fix multipart index regressions

1) The current keymanager in use index, was not being reset
for the next execution of a child query.
2) Index read with a keyed limit, gets the key count upfront.
The rest of the index read uses a key merger. The merger doesn't
handle getCount properly and uses a key manager for each key part
to aggregate the count. When resetting the key manager, it needs
to avoid resetting the unused key merger manager.
3) A few problems with eoi in derived activities, e.g. IndexCount
4) IndexCount getCount() usage fix, can't use checkCount

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
Merge pull request #8725 from richardkchapman/clone-reset-results

HPCC-15494 Workunit results aren't cleared when WU is cloned

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #8729 from wangkx/h15664

HPCC-15664 Add gzip header to http/soapcall test result key files

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jamienoss committed:
HPCC-15210 Compile direct from .gz files

Signed-off-by: James Noss <james.noss@lexisnexis.com>

Commit:
HPCC-14168 Include who aborted a WU

Include who aborted a work unit on the work unit details page.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

Michael-Gardner committed:
HPCC-15671 Dpkg-shlibdeps fix for cassandra library

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

richardkchapman committed:
Merge pull request #8735 from Michael-Gardner/HPCC-15671

HPCC-15671 Dpkg-shlibdeps fix for cassandra library

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8732 from jakesmith/hpcc-15666

HPCC-15666 Fix multipart index regressions

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-15286 Improve eclccserver tracing

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

rpastrana committed:
HPCC-15619 Code review changes

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

jakesmith committed:
HPCC-15678 Index group aggregate on multi slave thor deadlock

Regression/incomplete fix in HPCC-15666

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
Merge pull request #8739 from jakesmith/hpcc-15678

HPCC-15678 Index group aggregate on multi slave thor deadlock

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #8724 from rpastrana/HPCC-15619Enhancegetbinding

HPCC-15619 Enhance WsESDLConfig getESDLBinding method

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8599 from ghalliday/issue15362

HPCC-15362 Remove other uses of atomic_t from jlib

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8734 from miguelvazq/HPCC-14168

HPCC-14168 Include who aborted a WU

Reviewed-By: Gordon Smith <gordon.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8581 from shamser/issue13826

HPCC-13826 Extension of index_field[1..n]=field to IN

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #8671 from jamienoss/issue14405-redis-plugin-cache-all-sub-connections-rewrite

HPCC-14405 Redis Plugin - use cached connection for subscriptions

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #8738 from richardkchapman/eclccserver-tracing

HPCC-15286 Improve eclccserver tracing

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ericsuhong committed:
HPCC-15702 Misspelled property name in ws_ecl_service.cpp

Fixed a misspelled attribute (@protalurl -> @portalurl)

Signed-off-by: Suk Hwan Hong <suk.hong@gatech.edu>

richardkchapman committed:
Merge pull request #8756 from sukhong/HPCC-15702

HPCC-15702 Misspelled property name in ws_ecl_service.cpp

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-15705 Fix regression in HPCC-15362

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #8758 from ghalliday/issue15705

HPCC-15705 Fix regression in HPCC-15362

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jamienoss committed:
HPCC-15710 Redis Plugin - close connection on exception

Signed-off-by: James Noss <james.noss@lexisnexis.com>

richardkchapman committed:
HPCC-14405 Redis Plugin - cache all subscription connections

Fix build error on CLang

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Build error on OSX

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8768 from richardkchapman/redis-build-error

HPCC-14405 Redis Plugin - cache all subscription connections

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

shamser committed:
HPCC-15701 Add no_likely

Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

richardkchapman committed:
Merge pull request #8769 from richardkchapman/esdlconfig-build-error

Build error on OSX

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-14988 Projecting fields into embeds

Also fixes HPCC-15724 (support non-constant embedded code), and adds the
ability to specify attributes on the EMBED(language, expr) form of the syntax.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-13194 Adding subfiles to super via ECLwatch

Simplifying the way users can add to superfiles or create new ones via the interface.

Signed-off-by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

richardkchapman committed:
Merge pull request #8782 from miguelvazq/HPCC-13194

HPCC-13194 Adding subfiles to super via ECLwatch

Reviewed-By: Gordon Smith <gordon.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8785 from gaojx/h15657_logging_filter_type2

HPCC-15657: ESPLogging Agent: Filters Type should be enumeration

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-14988 Projecting fields into embeds

Changes following review.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8784 from jamienoss/issue15710-redis-plugin-close-connection-on-thrown-exception

HPCC-15710 Redis Plugin - close connection on exception

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8776 from richardkchapman/project-embed-variable

HPCC-14988 Projecting fields into embeds

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #8778 from jamienoss/issue15210-directly-compile-gzipped-archives

HPCC-15210 Compile direct from .gz files

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

jamienoss committed:
HPCC-15616 Redis plugin - UNWATCH WATCHed key in UNLOCK

Signed-off-by: James Noss <james.noss@lexisnexis.com>

richardkchapman committed:
HPCC-15757 Add non-inplace string replace function to StringBuffer

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jamienoss committed:
HPCC-15760 __declspec(noreturn) needed with __attribute__((noreturn))

Signed-off-by: James Noss <james.noss@lexisnexis.com>

shamser committed:
HPCC-15749 Add new EPlikelihood property

Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

ghalliday committed:
Merge pull request #8801 from jamienoss/issue15760-missing-declspec-noreturn-from-plugin-code

HPCC-15760 __declspec(noreturn) needed with __attribute__((noreturn))

ghalliday committed:
Merge pull request #8799 from richardkchapman/optimize-string-replace

HPCC-15757 Add non-inplace string replace function to StringBuffer

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #8777 from shamser/issue15701

HPCC-15701 Add no_likely

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-14988 Projecting fields into embeds

Extra trailing nulls from confusion between utf8 chars versus bytes.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8798 from jamienoss/issue15616-redis-plugin-UNWATCH-WATCHed-key-in-UNLOCK

HPCC-15616 Redis plugin - UNWATCH WATCHed key in UNLOCK

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-14988 Projecting fields into embeds

Cleaner version of the fix

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8802 from richardkchapman/subst-string-utf8

HPCC-14988 Projecting fields into embeds

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
HPCC-15682 Add auto project for unprojected indexes

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

Commit:
HPCC-15590 Refer users to admin for group rename

Fixes HPCC-15590

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

richardkchapman committed:
HPCC-15727 Can't fold functions in eclrtl on OSX

Something of a hack, but helps keep OSX compiler behaviour consistent with
linux and will save me confusion in the futore.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8736 from miguelvazq/HPCC-15590

HPCC-15590 Unable to rename group in summary tab

Reviewed-By: Gordon Smith <gordon.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8804 from shamser/issue15747c

HPCC-15749 Add new EPlikelihood property

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

shamser committed:
HPCC-15750 Implement getLikelihood for And, OR, NOT

Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

mckellyln committed:
HPCC-15788 Thor logs should report virtual slave info

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

ghalliday committed:
Merge pull request #8811 from richardkchapman/osx-eclrtl-fold

HPCC-15727 Can't fold functions in eclrtl on OSX

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-15793 Cassandra plugin not being built

Code was testing old CMake variables that are no longer set anywhere.

Also fixed the cassandra logging agent to link to the cassandraembed plugin
rather than including one of its sources.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.4'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-15798 Aborting Workunit Translation missing

Fixes HPCC-15798

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

richardkchapman committed:
Merge pull request #8824 from miguelvazq/HPCC-15798

HPCC-15798 Aborting Workunit Translation missing

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

mckellyln committed:
HPCC-15788 Thor logs should report virtual slave info

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

ghalliday committed:
Merge pull request #8822 from richardkchapman/cassandra-build

HPCC-15793 Cassandra plugin not being built

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #8810 from ghalliday/issue15682

HPCC-15682 Add auto project for unprojected indexes

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

shamser committed:
HPCC-15750 Implement getLikelihood for And, OR, NOT

Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

mckellyln committed:
HPCC-12883 Remove socket fd max from select and epoll

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest 
Sha: f745f019293eedf3e5379ee8a2f2f96126b76177
Build: success
Install hpccsystems-platform-community_6.0.3-closedown0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest 
Sha: 09ec43ed277a8cb2ed05a203fe6a276ec4f358fb
Build: success
Install hpccsystems-platform-community_6.0.3-closedown0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

ghalliday committed:
Merge pull request #8830 from shamser/issue15750

HPCC-15750 Implement getLikelihood for And, OR, NOT, optimize nested likely

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

jakesmith commented on system/jlib/jsocket.cpp:
```diff
@@ -3744,29 +3772,42 @@ class CSocketBaseThread: public Thread
+                StringBuffer sockstr;
+                const char *tracename = sockstr.append((unsigned)sock).str();
+                LOGERR2(ERRNO(),3,"CSocketBaseThread poll handle");
+                return false;
```
trivial: would be nice if the above 2 identical block were common up.

mckellyln replied:
Done.

jakesmith commented on system/jlib/jsocket.cpp:
```diff
@@ -4316,19 +4396,37 @@ class CSocketSelectHandler: public CInterface, implements ISocketSelectHandler
+            DBGLOG("EPOLL: op(%d) bad fd %d to epfd %d", op, fd, efd);
+# endif
+            return;
+        }
```
Are the above conditions possible?

mckellyln replied:
ok, probably not - removed.

jakesmith commented on system/jlib/jsocket.cpp:
```diff
@@ -3965,6 +4006,45 @@ class CSocketSelectThread: public CSocketBaseThread
+    bool checkSocks()
+    {
+        bool ret = false;
+        ForEachItemIn(i,items) {
```
formatting: block on newline

(A few more below)

mckellyln replied:
Change in many places, but left select method code as is.

jakesmith replied:
Ok, my policy is that if I touch a method I correct the formatting at that point. That sometimes extends to if I touch a class.

jakesmith commented on system/jlib/jsocket.cpp:
```diff
@@ -4415,7 +4514,8 @@ class CSocketEpollThread: public CSocketBaseThread
         offset = 0;
         selecttrace = trc;
-        epfd = ::epoll_create(XFD_SETSIZE);
+        max_ret_events = 5000;
```
minor: perhaps should be a constant defined elsewhere (e.g. at head of file), with a comment describing significance.

jakesmith commented on system/jlib/jsocket.cpp:
```diff
@@ -4447,14 +4537,19 @@ class CSocketEpollThread: public CSocketBaseThread
+                epoll_op(epfd, EPOLL_CTL_DEL, si, 0);
+                si->nfy->Release();
+                si->sock->Release();
+                delete si;
```
It might be cleaner here and in other places, if a ~SelectItem handled releasing it's linked items (nfy, sock)

jakesmith replied:
It might also be worth making SelectItem a owned object and have the items array own them for tidier clearup.

mckellyln replied:
I agree SelectItem dtor could do nfy/sock Release but I was hesitant to change any of the select method code.

jakesmith replied:
fair enough. Could be another PR anyway.

jakesmith commented on system/jlib/jsocket.cpp:
```diff
@@ -4476,105 +4570,116 @@ class CSocketEpollThread: public CSocketBaseThread
-        sn.del = false;
-        sn.add_epoll = true;
+        CriticalBlock block(sect);
+        if (sock==NULL) { // wait until no changes outstanding
```
formatting: brace on newline.

jakesmith commented:
@mckellyln - looks good, minor comments only.

mckellyln committed:
HPCC-12883 Remove socket fd max from select and epoll

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

mckellyln commented:
@jakesmith thanks, addressed most of your comments with additional commit.

HPCCSmoketest commented:
Automated Smoketest 
Sha: 5d180f0af9b0930ea3427c8138c67b1104694b7e
Build: success
Install hpccsystems-platform-community_6.0.3-closedown0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

mckellyln committed:
HPCC-12883 Remove socket fd max from select and epoll

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

mckellyln commented:
@jakesmith I've removed the extra checks in epoll_op() and I've changed a few more single link block start formatting for the methods in CSocketEpollThread class.

HPCCSmoketest commented:
Automated Smoketest 
Sha: cd7e6b4698b36836f8333e0d8dc1f6eda008df0c
Build: success
Install hpccsystems-platform-community_6.0.3-closedown0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest 
Sha: cd7e6b4698b36836f8333e0d8dc1f6eda008df0c
Build: success
Install hpccsystems-platform-community_6.0.3-closedown0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

jakesmith commented:
Looks good, but should this be a 6.2 change not a point change?
Not needed for 6.0.x (afaik) and _may_ bring with it unknown issues potentially.
@richardkchapman

richardkchapman commented:
Feels like a 6.2 change to me

mckellyln commented:
closing to retarget 6.2.0 (master)
